src/dbnode/client/host_queue.go

```diff
@@ -879,6 +885,20 @@ func (q *queue) asyncFetchTagged(op *fetchTaggedOp) {
 		ctx, _ := thrift.NewContext(q.opts.FetchRequestTimeout())
 		result, err := client.FetchTagged(ctx, &op.request)
 		if err != nil {
+			if se, ok := err.(tchannel.SystemError); ok {
+				if se.Code() == tchannel.ErrCodeUnexpected &&
+					strings.Contains(se.Message(), rpc.ErrorType_BAD_REQUEST.String()) {
+
+					newErr, parseErr := tterrors.ParseErrorFromString(se.Message())
+					if parseErr == nil {
+						err = newErr
+					} else {
+						q.logger.Debug("failed to parse application error from error string. " +
```

Comment:
should this be higher than debug? seems like an issue

Reply:
Good call. If we get a bad request from our server and can't parse it, something has gone sideways. Bumped to a warn.